javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java

```diff
@@ -56,69 +56,83 @@ public final void write(ConvenienceMethod convenienceMethodObj, JavaClass classB
-                                switch (parameter.getProxyMethodParameter().getRequestParameterLocation()) {
-                                    case HEADER:
-                                        writeHeader(parameter, methodBlock);
-                                        break;
-
-                                    case QUERY:
-                                        writeQueryParam(parameter, methodBlock);
-                                        break;
-
-                                    case BODY: {
-                                        Consumer<JavaBlock> writeLine = javaBlock -> javaBlock.line(
-                                                String.format("requestOptions.setBody(%s);",
-                                                        expressionConvertToBinaryData(parameter.getName(), parameter.getClientMethodParameter().getClientType())));
-                                        if (!parameter.getClientMethodParameter().getIsRequired()) {
-                                            methodBlock.ifBlock(String.format("%s != null", parameter.getName()), ifBlock -> {
-                                                writeLine.accept(ifBlock);
-                                            });
-                                        } else {
-                                            writeLine.accept(methodBlock);
-                                        }
-                                    }
-                                    break;
-                                }
-                            }
-                        }
 
-                        // invocation with protocol method parameters and RequestOptions
-                        String invocationExpression = protocolMethod.getMethodInputParameters().stream()
-                                .map(p -> {
-                                    String expression = parameterExpressionsMap.get(p.getName());
-                                    if (expression == null) {
-                                        expression = p.getName();
-                                    }
-                                    return expression;
-                                })
-                                .collect(Collectors.joining(", "));
-
-                        // write the invocation of protocol method, and related type conversion
-                        writeInvocationAndConversion(convenienceMethod, protocolMethod, invocationExpression, methodBlock);
+                        writeMethodImplementation(protocolMethod, convenienceMethod, methodBlock);
                     });
                 });
     }
 
+    /**
+     * Write the implementation of the convenience method.
+     *
+     * @param protocolMethod the protocol method.
+     * @param convenienceMethod the convenience method.
+     * @param methodBlock the code block.
+     */
+    protected void writeMethodImplementation(
```

Comment:
Modification on this class is refactor. To make subclass can override `writeMethodImplementation`.